File: src/2574-Left-and-Right-Sum-Differences/left_and_right_sum_differences.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> leftRightDifference(std::vector<int>& nums) {
        int left = 0;
        int right = 0;
        std::vector<int> ans = {};

        for (auto n : nums) {
            right += n;
        }

        for (auto n : nums) {
            right -= n;
            ans.push_back(std::abs(left - right));
            left +=n;
        }

        return ans;
    }
};

int main() {
    std::vector<int> nums = {10,4,8,3};
    Solution solution;

    std::vector<int> output =  solution.leftRightDifference(nums);

    return 0;
}

File: src/2574-Left-and-Right-Sum-Differences/left_and_right_sum_differences.py
from typing import List

class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        left: int = 0
        right: int = 0
        ans: List[int] = []

        for n in nums:
            right += n

        for n in nums:
            right -= n
            ans.append(abs(left - right))
            left += n
            
        return ans

if __name__ == "__main__":
    nums: List[int] = [10,4,8,3]
    solution = Solution()

    print(solution.leftRightDifference(nums))